semanticdb/semanticdb3/guide.md

```diff
@@ -0,0 +1,576 @@
+command-line arguments as `scalac` and automatically installs
+`semanticdb-scalac` into the compilation pipeline, saving its users the hassle
+of configuring compiler plugins. Metac is ideal for quick experiments
+with SemanticDB.
+
+```bash
+alias metac="coursier launch org.scalameta:metac_2.11:3.7.4 -- -cp $(coursier fetch -p org.scala-lang:scala-library:2.11.12)"
+```
+
+For an example of using Metac, check out [SemanticDB example](#semanticdb-example).
+
+### Sbt
+
+In order to enable `semanticdb-scalac` for your sbt project, add the following
+to your build. Note that the compiler plugin requires the -Yrangepos compiler
+option to be enabled.
+
+```scala
+addCompilerPlugin("org.scalameta" % "semanticdb-scalac" % "3.7.4" cross CrossVersion.full)
+scalacOptions += "-Xplugin-require:semanticdb"
+scalacOptions += "-Yrangepos"
+```
+
+### Metacp
+
+Metacp is a command-line tool that takes a classpath, dumps its metadata in
+SemanticDB format and returns a classpath of SemanticDB files.
+Advanced command-line options control caching, parallelization
+and interaction with some quirks of the Scala standard library.
+
+```bash
+alias metacp="coursier launch org.scalameta:metacp_2.11:3.7.4 --"
+```
+
+As an example of using Metacp, let's compile Test.scala from
+[SemanticDB example](#semanticdb-example) using Scalac and then convert
+the resulting classfiles to SemanticDB. Note that newer versions of Metac
+may also generate an accompanying .semanticidx file, but we won't be discussing
+it in this document.
+
+```
+$ scalac Test.scala
+<success>
+
+$ tree
+.
+├── Test$.class
+├── Test.class
+└── Test.scala
+
+$ metacp .
+/var/folders/pg/v06175sd2_qb9jbg7c28xr040000gn/T/semanticdb8798662205463788559
+
+$ tree /var/folders/pg/v06175sd2_qb9jbg7c28xr040000gn/T/semanticdb8798662205463788559
+/var/folders/pg/v06175sd2_qb9jbg7c28xr040000gn/T/semanticdb8798662205463788559
+└── META-INF
+    └── semanticdb
+        └── Test.class.semanticdb
+
+$ metap /var/folders/pg/v06175sd2_qb9jbg7c28xr040000gn/T/semanticdb8798662205463788559
+Test.class
+----------
+
+Summary:
+Schema => SemanticDB v3
+Uri => Test.class
+Text => empty
+Language => Scala
+Symbols => 5 entries
+
+Symbols:
+_empty_. => package _empty_
+_empty_.Test. => final object Test.{+2 decls}
+  extends AnyRef
+_empty_.Test.main(Array). => method main: (args: Array[String]): Unit
+  args => _empty_.Test.main(Array).(args)
+  Array => scala.Array#
+  String => scala.Predef.String#
+  Unit => scala.Unit#
+_empty_.Test.main(Array).(args) => param args: Array[String]
+  Array => scala.Array#
+  String => scala.Predef.String#
+_root_. => package _root_
+```
+
+Note how the resulting SemanticDB payload is missing the `Text` attribute and
+the `Occurrences` section. That's because Metacp doesn't know anything about
+sources and only works with classes.
+
+<table>
+  <tr>
+    <td width="140px">Option</td>
+    <td>Value</td>
+    <td>Explanation</td>
+    <td>Default</td>
+  </tr>
+  <tr>
+    <td></td>
```

Comment:
Empty "Value" is a bit weird, should we maybe write `<classpath>` and prefix this table with usage syntax?

```
Usage: metacp [options] <classpath>
```